sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/client.py

```diff
@@ -204,6 +208,10 @@ def __init__(self, hostname, **kwargs):
                 "Http proxy settings can't be passed if transport_type is explicitly set to Amqp"
             )
         self._transport_type = kwargs.pop("transport_type", TransportType.Amqp)
+        if self._transport_type.value == TransportType.Amqp.value:
+            self._socket_timeout = kwargs.pop("socket_timeout", SOCKET_TIMEOUT)
+        else:
+            self._socket_timeout = kwargs.pop("socket_timeout", WS_TIMEOUT_INTERVAL)
```

Comment:
This default-setting behaviour should be done in the Connection object where it's used, rather than the client.